### PR TITLE
JSUI-2991 Sort suggestions by index

### PR DESCRIPTION
**Pull request by ThibodeauJF**
That functionality was lost in a previous refactor.

https://coveord.atlassian.net/browse/JSUI-2991

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/magicbox/SuggestionsManager.ts [hunk 1]
```diff
@@ -186,7 +186,7 @@ export class SuggestionsManager {
       return;
     }
 
-    each(suggestions, (suggestion: Suggestion) => {
+    suggestions.sort((a, b) => b.index || 0 - a.index || 0).forEach(suggestion => {
```
Comment (samisayegh) [inline src/magicbox/SuggestionsManager.ts:189]: Hmm, I think we could simplify the sort function here. If descending order by index, then:
```
suggestions.sort((a,b) => b.index - a.index)
```

Comment (ThibodeauJF) [reply src/magicbox/SuggestionsManager.ts:189]: won't work if an index is undefined

Comment (samisayegh) [reply src/magicbox/SuggestionsManager.ts:189]: I see, I think a test capturing this case would be useful if we do not have one already.

Comment (ThibodeauJF) [reply src/magicbox/SuggestionsManager.ts:189]: In the current test I only added a value to one of the suggestion, that's how I catched that issue 😂

**Pull request merged**